dev/maven-plugin/src/main/scala/com/lightbend/lagom/maven/MavenFacade.scala

```diff
@@ -0,0 +1,276 @@
+package com.lightbend.lagom.maven
+
+import java.io.File
+import java.util
+import java.util.Collections
+import javax.inject.{ Inject, Singleton }
+
+import org.apache.maven.RepositoryUtils
+import org.apache.maven.artifact.ArtifactUtils
+import org.apache.maven.execution.MavenSession
+import org.apache.maven.lifecycle.internal._
+import org.apache.maven.model.Plugin
+import org.apache.maven.plugin.{ BuildPluginManager, MojoExecution }
+import org.apache.maven.project.MavenProject
+import org.codehaus.plexus.configuration.PlexusConfiguration
+import org.codehaus.plexus.util.StringUtils
+import org.codehaus.plexus.util.xml.Xpp3Dom
+import org.eclipse.aether.{ DefaultRepositorySystemSession, RepositorySystem }
+import org.eclipse.aether.artifact.Artifact
+import org.eclipse.aether.collection.CollectRequest
+import org.eclipse.aether.graph.Dependency
+import org.eclipse.aether.repository.{ WorkspaceReader, WorkspaceRepository }
+import org.eclipse.aether.resolution.{ DependencyRequest, DependencyResult }
+
+import scala.collection.JavaConverters._
+
+/**
+ * Facade in front of maven.
+ *
+ * All the hairy stuff goes here.
+ */
+@Singleton
+class MavenFacade @Inject() (repoSystem: RepositorySystem, session: MavenSession,
+    buildPluginManager: BuildPluginManager, lifecycleExecutionPlanCalculator: LifecycleExecutionPlanCalculator) {
+
+  val cachedProjectIndex = new ProjectIndex(session.getProjects)
```

Comment:
You're right - it's a remnant from when I was trying to do some things a different way.